rust/src/asn1/parse_rules.rs

```diff
@@ -0,0 +1,270 @@
+                }
+            })
+        },
+    )(input)
+}
+
+/// Parse asn1 keyword options
+pub(super) fn asn1_parse_rule(input: &str) -> IResult<&str, DetectAsn1Data> {
+    // If nothing to parse, return
+    if input.is_empty() {
+        return Err(nom::Err::Error(nom::error::make_error(
+            input,
+            nom::error::ErrorKind::Eof,
+        )));
+    }
+
+    // Rule parsing functions
+    fn bitstring_overflow(i: &str) -> IResult<&str, &str> {
+        tag("bitstring_overflow")(i)
+    }
+
+    fn double_overflow(i: &str) -> IResult<&str, &str> {
+        tag("double_overflow")(i)
+    }
+
+    fn oversize_length(i: &str) -> IResult<&str, (&str, u32)> {
+        separated_pair(tag("oversize_length"), multispace1, parse_u32_number)(i)
+    }
+
+    fn absolute_offset(i: &str) -> IResult<&str, (&str, u16)> {
+        separated_pair(tag("absolute_offset"), multispace1, parse_u16_number)(i)
+    }
+
+    fn relative_offset(i: &str) -> IResult<&str, (&str, i32)> {
+        separated_pair(
+            tag("relative_offset"),
+            multispace1,
+            verify(parse_i32_number, |v| {
+                *v >= -1 * i32::from(std::u16::MAX) && *v <= i32::from(std::u16::MAX)
+            }),
+        )(i)
+    }
+
+    let mut data = DetectAsn1Data::default();
+
+    let mut rest = input;
+
+    // Parse the input and set data
+    while !rest.is_empty() {
+        let (
+            new_rest,
+            (
+                _,
+                bitstring_overflow,
+                double_overflow,
+                oversize_length,
+                absolute_offset,
+                relative_offset,
+                _,
+            ),
+        ) = tuple((
+            opt(multispace0),
+            opt(bitstring_overflow),
+            opt(double_overflow),
+            opt(oversize_length),
+            opt(absolute_offset),
+            opt(relative_offset),
+            opt(alt((multispace1, tag(",")))),
+        ))(rest)?;
+
+        if bitstring_overflow.is_some() {
+            data.bitstring_overflow = true;
+        } else if double_overflow.is_some() {
+            data.double_overflow = true;
+        } else if let Some((_, v)) = oversize_length {
+            data.oversize_length = Some(v);
+        } else if let Some((_, v)) = absolute_offset {
+            data.absolute_offset = Some(v);
+        } else if let Some((_, v)) = relative_offset {
+            data.relative_offset = Some(v);
+        } else {
+            return Err(nom::Err::Error(nom::error::make_error(
+                rest,
+                nom::error::ErrorKind::Verify,
+            )));
+        }
+
+        rest = new_rest;
+    }
+
+    Ok((rest, data))
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use test_case::test_case;
+
+    // Test oversize_length
+    #[test_case("oversize_length 1024", DetectAsn1Data { oversize_length: Some(1024), ..Default::default()}; "check that we parse oversize_length correctly")]
```

Comment:
can you update these to not exceed line length limit? ~80 char soft limit, 100 hard limit